app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -446,14 +446,22 @@ def smart_proxy_service
       def rhsm_url
         # Since Foreman 3.1 this setting is set
         if (rhsm_url = setting(SmartProxy::PULP3_FEATURE, 'rhsm_url'))
-          rhsm_url
+          URI(rhsm_url)
```

Comment:
I'm not sure of the impact overall of this change. Why was it necessary for this PR as my assumption this has been returning a string for some time

Reply:
I see that you're using the URI object in the template. Got it